react/features/device-selection/components/AudioOutputPreview.js

```diff
@@ -0,0 +1,146 @@
+        /**
+         * The device id of the audio output device to use.
+         */
+        deviceId: React.PropTypes.string,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    }
+
+    /**
+     * Initializes a new AudioOutputPreview instance.
+     *
+     * @param {Object} props - The read-only React Component props with which
+     * the new instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this._audioEl = null;
+
+        this._onClick = this._onClick.bind(this);
+        this._setAudioEl = this._setAudioEl.bind(this);
+    }
+
+    /**
+     * Sets the target output device on the component's audio element
+     * after initial render.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        this._setSink();
+    }
+
+    /**
+     * Updates the audio element when the target output device changes and
+     * the audio element has re-rendered.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidUpdate() {
+        this._setSink();
+    }
+
+    /**
+     * Ensures the audio element no longer has a sound to play.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        this._stopAudioEl();
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <div className = 'audio-output-preview'>
+                <a onClick = { this._onClick }>
+                    { this.props.t('deviceSelection.testAudio') }
+                </a>
+                <audio
+                    preload = 'auto'
+                    ref = { this._setAudioEl }
+                    src = { testSoundFilePath } />
+            </div>
+        );
+    }
+
+    /**
+     * Plays a test sound.
+     *
+     * @returns {void}
+     */
+    _onClick() {
+        this._audioEl && this._audioEl.play();
+    }
+
+    /**
+     * Updates the target output device for playing the test sound.
+     *
+     * @returns {void}
+     * @private
+     */
+    _setSink() {
+        this._audioEl
+        && this._audioEl.setSinkId
+        && this._audioEl.setSinkId(this.props.deviceId);
+    }
+
+    /**
+     * Sets the instance variable for the component's audio element
+     * so it can be accessed directly.
+     *
+     * @param {Object} el - The DOM element for the component's audio.
+     * @returns {void}
+     * @private
+     */
+    _setAudioEl(el) {
+        this._stopAudioEl();
+        this._audioEl = el;
+    }
+
+    /**
+     * Stops the component's audio element from playing sounds.
+     *
+     * @returns {void}
+     * @private
+     */
+    _stopAudioEl() {
+        if (this._audioEl) {
+            this._audioEl.pause();
+            this._audioEl.removeAttribute('src');
+            this._audioEl.load();
```

Comment:
Thinking about it more, maybe not. I was initially inspired to put in this cleanup because of previous audio switching I did for another project but the use case was different. I've now removed this logic.